design-proposals/pvDiskStorage.md

```diff
@@ -0,0 +1,132 @@
+# Using Persistent Volumes as Virtual Machine disks
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+Virtual Machines use to have disks attached. They are not always required, but
+have some value if you need to persist data.
+
+Kubernetes provides persistent storage through Persistent Volumes and
+Claims.
+
+The purpose of this proposal is to describe a mechanism to use Persistent
+Volumes as a backing store for Virtual Machine disks.
+
+
+### Use-case
+
+The primary use-case is to attach regular (writable) disks to Virtual Machines
+which are backed by Peristent Volumes.
+
+
+## API
+
+This section is concerned about how the Persistent Volumes are referenced
+in the `VM` Resource type.
+
+In general the referencing is aligned with how pods are consuming Persistent
+Volume Claims as described [here](https://kubernetes.io/docs/api-reference/v1.5/#persistentvolumeclaimvolumesource-v1)
+
+Today the `VM.spec.domain` reflects much of
+[libvirt's domain xml specification](http://libvirt.org/formatdomain.html#elementsDisks).
+To support Persistent Volumes as a backing store for file storage, two changea are
+proposed:
+
+1. an additional disk type `PersistentVolumeClaim` is accepted
+   If this type is used, then
+2. an additional `claimName` attribute needs to be specified
+   which is specifying the claim to use
+
+Example with the following PV and PVC:
+
+```yaml
+kind: PersistentVolume
+metadata:
+  name: pv001
+  labels:
+    release: "stable"
+spec:
+  capacity:
+    storage: 5Gi
+  iscsi:
+    targetPortal: example.com:3260
+    iqn: iqn.2013-07.com.example:iscsi-nopool/
+    lun: 0
+---
+kind: PersistentVolumeClaim
+metadata:
+  name: disk-01
+spec:
+  resource:
+    requests:
+      storage: 4Gi
+  selector:
+    matchLabels:
+release: "stable"
+```
+
+this is used by the Vm in the following way:
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      disks:
+      - type: PersistentVolumeClaim
```

Comment:
Is this the complete set of fields?  Document should have a section that describes each of the acceptable  fields and where they map to on dom.xml or in the k8s system